core/src/main/resources/error/error-classes.json

```diff
@@ -1,200 +1,200 @@
+    "message" : [ "PARTITION clause cannot contain a non-partition column name: <columnName>" ],
     "sqlState" : "42000"
   },
   "NO_HANDLER_FOR_UDAF" : {
-    "message" : [ "No handler for UDAF '%s'. Use sparkSession.udf.register(...) instead." ]
+    "message" : [ "No handler for UDAF 'functionName'. Use sparkSession.udf.register(...) instead." ]
   },
   "NO_UDF_INTERFACE_ERROR" : {
-    "message" : [ "UDF class %s doesn't implement any UDF interface" ]
+    "message" : [ "UDF class <class> doesn't implement any UDF interface" ]
   },
   "PARSE_CHAR_MISSING_LENGTH" : {
-    "message" : [ "DataType %s requires a length parameter, for example %s(10). Please specify the length." ],
+    "message" : [ "DataType <type> requires a length parameter, for example <type>(10). Please specify the length." ],
     "sqlState" : "42000"
   },
   "PARSE_EMPTY_STATEMENT" : {
     "message" : [ "Syntax error, unexpected empty statement" ],
     "sqlState" : "42000"
   },
   "PARSE_SYNTAX_ERROR" : {
-    "message" : [ "Syntax error at or near %s%s" ],
+    "message" : [ "Syntax error at or near <error><hint>" ],
     "sqlState" : "42000"
   },
   "PIVOT_VALUE_DATA_TYPE_MISMATCH" : {
-    "message" : [ "Invalid pivot value '%s': value data type %s does not match pivot column data type %s" ],
+    "message" : [ "Invalid pivot value '<value>': value data type <valueType> does not match pivot column data type <pivotType>" ],
     "sqlState" : "42000"
   },
   "RENAME_SRC_PATH_NOT_FOUND" : {
-    "message" : [ "Failed to rename as %s was not found" ],
+    "message" : [ "Failed to rename as <sourcePath> was not found" ],
     "sqlState" : "22023"
   },
   "SECOND_FUNCTION_ARGUMENT_NOT_INTEGER" : {
-    "message" : [ "The second argument of '%s' function needs to be an integer." ],
+    "message" : [ "The second argument of '<functionName>' function needs to be an integer." ],
     "sqlState" : "22023"
   },
   "UNABLE_TO_ACQUIRE_MEMORY" : {
-    "message" : [ "Unable to acquire %s bytes of memory, got %s" ]
+    "message" : [ "Unable to acquire <requestedBytes> bytes of memory, got <receivedBytes>" ]
   },
   "UNRECOGNIZED_SQL_TYPE" : {
-    "message" : [ "Unrecognized SQL type %s" ],
+    "message" : [ "Unrecognized SQL type <typeName>" ],
     "sqlState" : "42000"
   },
   "UNSUPPORTED_DATATYPE" : {
-    "message" : [ "Unsupported data type %s" ],
+    "message" : [ "Unsupported data type <typeName>" ],
     "sqlState" : "0A000"
   },
   "UNSUPPORTED_FEATURE" : {
-    "message" : [ "The feature is not supported: %s" ],
+    "message" : [ "The feature is not supported: <feature>" ],
     "sqlState" : "0A000"
   },
   "UNSUPPORTED_GROUPING_EXPRESSION" : {
     "message" : [ "grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup" ]
   },
   "UNSUPPORTED_OPERATION" : {
-    "message" : [ "The operation is not supported: %s" ]
+    "message" : [ "The operation is not supported: <operation>" ]
```

Comment:
@srielau it's not really biggie but I think the previous `%s` format follows Python style https://docs.python.org/3/library/string.html#format-examples.

What about we change `<operation>` -> `{operation}` that's also Python named argument mapping style?  I know we're in JSON, not Python but wondered if this makes a bit of more sense since the exiting one was already similar with Python's. FWIW, this is what Bazel does too (https://docs.bazel.build/versions/main/skylark/language.html#starlark-language) in their own language.

Reply:
This is a good point, we should reuse existing standards.

Reply:
I'm OK with `<abc>` if it's also a widely adopted standard

Reply:
I'm worried about the curlies in the the docs... We use curies for the syntax diagram...

Reply:
makes sense